components/user-mgt/org.wso2.carbon.user.mgt/src/main/java/org/wso2/carbon/user/mgt/listeners/UserManagementAuditLogger.java

```diff
@@ -96,11 +111,19 @@ public boolean doPreSetUserClaimValue(String userName, String claimURI, String c
 
         if (isEnable()) {
             JSONObject dataObject = new JSONObject();
+            String target = ListenerUtils.getEntityWithUserStoreDomain(userName, userStoreManager);
+            if (LoggerUtils.isLogMaskingEnable) {
+                Map<String, String> claims = new HashMap<>();
+                claims.put(claimURI, claimValue);
```

Comment:
instead of having to create a map here, we can introduce a separate util method to handle if there is only one claim